pkg/operator/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go

```diff
@@ -55,7 +55,8 @@ type clusterManagerController struct {
 	cache                resourceapply.ResourceCache
 	// For testcases which don't need these functions, we could set fake funcs
 	ensureSAKubeconfigs func(ctx context.Context, clusterManagerName, clusterManagerNamespace string,
-		hubConfig *rest.Config, hubClient, managementClient kubernetes.Interface, recorder events.Recorder) error
+		hubConfig *rest.Config, hubClient, managementClient kubernetes.Interface, recorder events.Recorder,
+		mwctrEnabled, addonManagerEnabled bool) error
```

Comment:
can we just use features? it is global variable, we do not need them as input

Reply:
for hosted mode, there might be multiple clusterManager CRs, if they have different feature gate settings, I think using the global variable is not appropriate.